Automation/SQL/azureDeploy.ps1

```diff
@@ -0,0 +1,43 @@
+# Creates resource group and Azure SQL logical server
+
+param(
+        # Login parameters
+        [string] [Parameter(Mandatory=$true)] $tenantId,          
+        [string] [Parameter(Mandatory=$true)] $applicationId,
+        [string] [Parameter(Mandatory=$true)] $secret,
+        [string] [Parameter(Mandatory=$true)] $subscriptionId,
+        # Azure SQL server parameters
+        [string] [Parameter(Mandatory=$true)] $location,
+        [string] [Parameter(Mandatory=$true)] $resourceGroupName,
+        [string] [Parameter(Mandatory=$true)] $serverName,
+        [string] [Parameter(Mandatory=$true)] $administratorLogin,
+        [SecureString] [Parameter(Mandatory=$true)] $administratorLoginPassword
+)
+
+# Log in and set the SubscriptionId in which to create these objects
+Import-Module ..\Login
+Login $tenantId $applicationId $secret $subscriptionId
+
+$pathToAzSqlTemplate = "./azureDeploy.json"
+
+# Check for or create a resource group
+$resourceGroupExists = (Get-AzResourceGroup -Name $resourceGroupName `
+-ErrorVariable notPresent -ErrorAction SilentlyContinue).ResourceGroupName `
+-eq $resourceGroupName 
+
+if (!$resourceGroupExists) {
+        New-AzResourceGroup -Name $resourceGroupName -Location "$location" -Force
+        Write-Host "Created resource group $resourceGroupName"
+} else {
+        Write-Host "Resource group already exists."
+}
+
+# Deploy template
+Write-host "Creating primary server..."
```

Comment:
I like the idea of write hosts as they let the user know what stage the script is at.